crypto/pqdsa/pqdsa.c

```diff
@@ -82,14 +82,69 @@ int PQDSA_KEY_set_raw_public_key(PQDSA_KEY *key, CBS *in) {
 }
 
 int PQDSA_KEY_set_raw_private_key(PQDSA_KEY *key, CBS *in) {
```

Comment:
This implementation feels very much like two functions conflated into one.  

Could there instead be a separate function for the seed, like `PQDSA_KEY_set_raw_seed`? Then have the callers (`pqdsa_priv_decode` and `EVP_PKEY_pqdsa_new_raw_private_key`?) make the decision about which to call based on the length of the data?

Reply:
I see! Reworked in [e506ba1](https://github.com/aws/aws-lc/pull/2157/commits/e506ba191fae8d8c7fccef6215355243ba04e263)/